src/components/Enum4Linux/Enum4Linux.tsx

```diff
@@ -1,42 +1,51 @@
-import { Button, LoadingOverlay, Stack, TextInput } from "@mantine/core";
+import { Button, Stack, TextInput, Switch } from "@mantine/core";
 import { useForm } from "@mantine/form";
-import { useCallback, useState } from "react";
+import { useCallback, useState, useEffect } from "react";
 import { CommandHelper } from "../../utils/CommandHelper";
 import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
-import { UserGuide } from "../UserGuide/UserGuide";
-import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
+import { RenderComponent } from "../UserGuide/UserGuide";
 import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
+import React from "react";
 
-const title = "Enum4Linux";
-const description_userguide =
-    "Enum4linux is a tool used for enumerating information from Windows and Samba operating systems. " +
-    "It is particularly useful for identifying the remote OS of a system and providing a list of the users " +
-    "and group memberships found within the system.\n\nOptions for the tool can be found at: " +
-    "https://www.kali.org/tools/enum4linux/\n\n" +
-    "Using Enum4Linux:\n" +
-    "Step 1: Enter a target IP address. Eg: 192.168.1.1\n\n" +
-    "Step 2: Enter an option for the enumeration. Eg: U (get userlist)\n\n" +
-    "Step 3: Enter any parameters. Eg: example.txt\n\n" +
-    "Step 4: Enter any additional options/parameters.\n\n" +
-    "Step 5: Click Scan to commence Enum4Linux's operation.\n\n" +
-    "Step 6: View the output block below to view the results of the tools execution.";
-
-interface FormValues {
+/**
+ * Represents the form values for the Enum4Linux component.
+ */
+interface FormValuesType {
     ipAddress: string;
     argumentMain: string;
     paramMain: string;
     argumentAlt: string;
     paramAlt: string;
 }
 
+/**
+ * The Enum4Linux component.
+ */
 const Enum4Linux = () => {
     const [loading, setLoading] = useState(false);
     const [output, setOutput] = useState("");
     const [pid, setPid] = useState("");
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false);
     const [allowSave, setAllowSave] = useState(false);
     const [hasSaved, setHasSaved] = useState(false);
+    const [opened, setOpened] = useState(false);
```

Comment:
Should only be open if the command is not available.
```suggestion
    const [opened, setOpened] = useState(!isCommandAvailable);
```